packages/js/product-editor/src/contexts/validation-context/use-validation.ts

```diff
@@ -33,11 +33,13 @@ export function useValidation< T >(
 		ref,
 		error: context.errors[ validatorId ],
 		isValidating,
-		async validate() {
+		async validate( newData?: Record< string, unknown > ) {
 			setIsValidating( true );
-			return context.validateField( validatorId ).finally( () => {
-				setIsValidating( false );
-			} );
+			return context
+				.validateField( validatorId, newData )
```

Comment:
should we validate the `newData` before to validate it? I see it sends the SyntheticBaseEvent event, for instance, when validating. 

<img width="1728" alt="Screenshot 2024-01-08 at 11 02 56" src="https://github.com/woocommerce/woocommerce/assets/77539/4fb25fe4-5c13-4521-9e62-7eae880a6772">